azurerm/internal/services/network/parse/network_connection_monitor.go

```diff
@@ -0,0 +1,40 @@
+package parse
+
+import (
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type NetworkConnectionMonitorId struct {
+	ResourceGroup string
+	WatcherId     string
```

Comment:
Now that we have the `WatcherName`, we do we need to export this?

Reply:
Because we have to use it to set watcher id to statfile. 
d.Set("network_watcher_id", id.WatcherId)

Reply:
In this case, can we opt in ID formatter for the network watcher, and use the `NewXXX()` function in place where you need to set the network watcher id?

Example: https://github.com/magodo/terraform-provider-azurerm/blob/azurerm_vpn_gateway_connection/azurerm/internal/services/network/vpn_gateway_connection_resource.go#L373-L373

Reply:
thanks. updated